src/Security/Provider/UserProvider.php

```diff
@@ -43,20 +43,25 @@ public function __construct(UserRepository $userRepository, RequestStack $reques
     }
 
     /**
-     * @throws UsernameNotFoundException
+     * @throws UserNotFoundException
      */
     public function loadUserByUsername(string $username): UserInterface
     {
         try {
             return $this->userRepository->findByUsernameWithAssignments($username);
         } catch (ORMException $exception) {
-            throw new UsernameNotFoundException(sprintf("Username '%s' does not exist", $username));
+            throw new UserNotFoundException(sprintf("Username '%s' does not exist", $username));
```

Comment:
I think it makes sense to keep it as it is, we could use `user` if it would be by id. If it's not by primary key, I think it's ok to keep it as it is. What do you think?

Reply:
It's tricky because we are looking for a User by the Username, so none of then exists :)
I'm ok the way it is, what just trying to be consistent with the new name of the exception